src/Intercom/Clients/VisitorsClient.cs

```diff
@@ -107,12 +107,12 @@ public Visitor Delete(Visitor visitor)
         {
             if (visitor == null)
             {
-                throw new ArgumentNullException("'visitor' argument is null.");
+                throw new ArgumentNullException(nameof(visitor));
             }
 
             if (String.IsNullOrEmpty(visitor.id))
             {
-                throw new ArgumentNullException("'visitor.id' argument is null.");
+                throw new ArgumentException("'visitor.id' argument is null.");
```

Comment:
Hi!  No, I don't think so.  In general, the best practice for `ArgumentNullException`'s are to pass in the name of the variable that is `null`.  If there is something else wrong with it, then I believe a different type of exception is supposed to be raised like `ArgumentException`.  I could be totally wrong though 😄